### DISCUSSION TITLE
How to invert pin HIGH / LOW behaviour in Homespan DEV_LED() function? (For relay control)

### DISCUSSION THREAD

**Discussion by FidelDworp** [HomeSpan/HomeSpan#584] [Q&A] [answered]
I am trying to make a dual relay control with the Homespan DEV_LED() function.
Below sketch works fine in homekit, but the light controls work opposite of what I need:

When I turn the switch ON in HomeKit, the pin 25 or 26 go HIGH, but I need them to go LOW to control the relays.

Questions:

- Does somebody know if I can invert this behaviour? (and if possible HOW?)
- Or is there another function I can/should use instead?

```cpp
#include "HomeSpan.h" 
#include "DEV_LED.h"     // Include this file, DEV_LED.h

void setup()
{
  Serial.begin(115200);
  homeSpan.begin(Category::Lighting,"HomeSpan LED");
  
  new SpanAccessory(); 
  
    new Service::AccessoryInformation(); 
      new Characteristic::Identify();                

    new DEV_LED(25);       // instantiates a DEV_LED Service on Pin 25

  new SpanAccessory(); 
  
    new Service::AccessoryInformation();    
      new Characteristic::Identify();                       
  
    new DEV_LED(26);       // instantiates a DEV_LED Service on Pin 26
}


void loop()
{
  homeSpan.poll();
}
```

**Comment by HomeSpan** [comment, marked as answer]
In the DEV_LED code, just flip the following line `digitalWrite(ledPin,power->getNewVal());` to  `digitalWrite(ledPin,1-power->getNewVal());`.  The key is that you want to output a LOW on the pin if the power changes to ON, rather output a HIGH, which is the normal case.

**Reply by FidelDworp** [reply]
Thanks for your quick reply! I could not find this solution myself...
But meanwhile I also tried to use the tablelamp example with 2 accessories operating both relays. (pins 25 & 26)
I try to test both solutions, but now I can't get my ESP32 to show up in Apple Homekit... :-(

**Reply by HomeSpan** [reply]
If you make changes to your sketch and the Home App is no longer showing the Accessory, it may have move to another room.  Check "Default Room" under the Home App to make sure.

If you are still having problems, the best is to reset and re-pair the device.  First type 'H' into the Serial Monitor - this causes HomeSpan to forget it is paired to HomeKit and it generates a new set of pairing and device codes so it will look like a new device to to HomeKit.  Next, re-pair the device via the Home App.
